src/main/scala/io/reactivesocket/tck/DSLTestSubscriber.scala

```diff
@@ -23,17 +23,22 @@ import org.json4s._
 import org.json4s.native.Serialization
 
 
-class DSLTestSubscriber(writer : PrintWriter, initData: String, initMeta: String, kind: String)
-  extends Subscriber[Payload] with Subscription {
+
+class DSLTestSubscriber(writer : PrintWriter, initData: String, initMeta: String, kind: String,
+  client: DSLTestClient = null) extends Subscriber[Payload] with Subscription {
 
   implicit val formats = Serialization.formats(NoTypeHints)
 
   private var id: UUID = null
   this.id = UUID.randomUUID
 
+  private var clientID: Int = ClientIDGen.currentCount()
+  if (client != null) clientID = client.getID
```

Comment:
the line above seems to generate a new client id each time, then sometimes discards it.  Is that deliberate?